core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -807,7 +810,7 @@ public InodeDirectory getRoot() {
           currentInodeDirectory.getId(), pathComponents[k], missingDirContext);
       if (currentInodeDirectory.isPinned() && !newDir.isPinned()) {
         newDir.setPinned(true);
-        newDir.setMediumTypes(new HashSet<>(currentInodeDirectory.getMediumTypes()));
+        newDir.setMediumTypes(currentInodeDirectory.getMediumTypes());
```

Comment:
No defensive copy is needed anymore, since everything is immutable. Actually, if the child inode just uses a ref to the parent dir's medium Set, that'd be even better cuz there's one less object off the table. But at the moment it doesn't seem to be the case.